packages/access-api/src/routes/validate-email.js

```diff
@@ -134,6 +139,93 @@ async function session(req, env) {
     req.query.ucan,
     delegation.capabilities[0].nb.key
   )
+  if (req.method.toLowerCase() === 'post') {
+    const accessSessionResult = await validator.access(delegation, {
+      capability: Access.session,
+      principal: Verifier,
+      authority: env.signer,
+    })
+    if (accessSessionResult.error) {
+      throw new Error(
+        `unable to validate access session: ${accessSessionResult.error}`
+      )
+    }
+    const accountDID = accessSessionResult.audience.did()
+    // @todo: use new ucanto `Account` instead
+    const accountOneOffKey = await ed25519.generate()
+    const account = accountOneOffKey.withDID(accountDID)
+
+    const agentPubkey = accessSessionResult.capability.nb.key
+
+    const update = await ucanto.delegate({
+      issuer: env.signer,
+      audience: account,
+      capabilities: [
+        {
+          with: env.signer.did(),
+          can: './update',
+          nb: {
+            key: agentPubkey,
+          },
+        },
+      ],
+    })
+    const attestKeyAuthorizesAccount = await ucanto.delegate({
+      issuer: env.signer,
+      audience: { did: () => agentPubkey },
+      capabilities: [
+        {
+          with: env.signer.did(),
+          can: 'ucan/attest',
+          nb: {
+            proof: update.cid,
```

Comment:
This should be the CID for the (unsigned) delegation from an account to the agent. This delegation should be replacement for the `update` above. More concretely if `update` was attesting to the authorization of the key, this attestation should be for the authorization of the specific delegation.

That is the core change in the spec, if previously we were attesting to key authorization which implied sudo privileges, in a new version we attest to the specific delegation authorization. Which means it could be sudo privileges or a very specific capabilities.

Reply:
Reading further down I think you could just remove this delegation.

Reply:
this is what's needed to make this work https://github.com/web3-storage/w3protocol/pull/460/files#diff-0cebcdde20e55de2911e90a385f368240bd9109fc54b074b89759628b695e88bR202